x/mongo/driver/legacy.go

```diff
@@ -19,4 +19,5 @@ const (
 	LegacyKillCursors
 	LegacyListCollections
 	LegacyListIndexes
+	LegacyHandshake
```

Comment:
Use "operation.Legacy == LegacyHandshake" so that the operation knows that the current operation should use a legacy handshake, as defined by the specifications: "[API version is not requested and loadBalanced: False](https://github.com/mongodb/specifications/blob/master/source/mongodb-handshake/handshake.rst#connection-handshake)", if this is the first message.